conan/tools/google/bazeldeps.py

```diff
@@ -490,8 +397,48 @@ def repository_name(self):
         """
         return self._root_package_info.repository_name
 
+    def get_full_libs_info(self):
+
+        def is_shared(cpp_info_):
+            return cpp_info_.type == PackageType.SHARED
+
+        full_libs_info = defaultdict(list)
+        cpp_info = self._dep.cpp_info
+        deduced_cpp_info = cpp_info.deduce_full_cpp_info(self._dep)
+        package_folder_path = self.package_folder
+        # Root
+        if len(cpp_info.libs) > 1:
+            for lib_name in cpp_info.libs:
+                virtual_component = deduced_cpp_info.components.pop(f"_{lib_name}")  # removing it!
+                full_libs_info["root"].append(
+                    _LibInfo(lib_name, is_shared(virtual_component),
+                         _relativize_path(virtual_component.location, package_folder_path),
+                         _relativize_path(virtual_component.link_location, package_folder_path))
+                )
```

Comment:
For me these all these `_LibInfo` creation does not read very clean. 

What do you think about one of these two other options?

- Add a method to create the _LibInfo, something like:

```
def create_lib_info(lib_name, cpp_info_):
    return _LibInfo(
        name=lib_name,
        is_shared=cpp_info_.type == PackageType.SHARED,
        lib_path=_relativize_path(cpp_info_.location, package_folder_path),
        import_lib_path=_relativize_path(cpp_info_.link_location, package_folder_path)
    )
```

- Use a _LibInfo class instead of namedtuples:

```
class LibInfo:
    def __init__(self, lib_name, cpp_info_, package_folder_path):
        self.name = lib_name
        self.is_shared = cpp_info_.type == PackageType.SHARED
        self.lib_path = _relativize_path(cpp_info_.location, package_folder_path)
        self.import_lib_path = _relativize_path(cpp_info_.link_location, package_folder_path)
```